src/shared/components/query/CancerStudySelector.tsx

```diff
@@ -1,391 +1,412 @@
+                                                }, 400);
+                                            }
+                                        }}
+                                    />
+
+                                </div>
+                            );
+
+                        }}
+                    </Observer>
+
+
+                </FlexRow>
+
+                <SectionHeader style={{display: 'none'}} promises={[this.store.cancerTypes, this.store.cancerStudies]}>
+                    Select Studies:
+                    {!!(!this.store.cancerTypes.isPending && !this.store.cancerStudies.isPending && !this.store.profiledSamplesCount.isPending) && (
+                        <Observer>
+                            {() => {
+                                let numSelectedStudies = expr(() => this.store.selectableSelectedStudyIds.length);
+                                let selectedCountClass = classNames({
+                                    [styles.selectedCount]: true,
+                                    [styles.selectionsExist]: numSelectedStudies > 0
+                                });
+                                return (
+                                    <span
+                                        className={selectedCountClass}
+                                        onClick={() => {
+                                            if (numSelectedStudies)
+                                                this.store.showSelectedStudiesOnly = !this.store.showSelectedStudiesOnly;
+                                        }}
+                                    >
+										<b>{numSelectedStudies}</b> studies selected
+										(<b>{this.store.profiledSamplesCount.result.all}</b> samples)
+									</span>
+                                );
+                            }}
+                        </Observer>
+                    )}
+                </SectionHeader>
+
+                <FlexRow className={styles.cancerStudySelectorBody}>
+                    <If condition={this.store.maxTreeDepth > 0}>
+                        <Then>
+                            <div className={styles.cancerTypeListContainer}>
+                                <this.CancerTypeList/>
+                            </div>
+                        </Then>
+                    </If>
+                    <div className={styles.cancerStudyListContainer} data-test='cancerTypeListContainer'>
+
+                        <div className="checkbox" style={{marginLeft: 19}}>
+                            <If condition={shownStudies.length > 0}>
+                                <If condition={!this.logic.mainView.isFiltered && !_.isEmpty(quickSetButtons)}>
+                                    <Then>
+                                        <div className={styles.quickSelect}>
+                                            <QuickSelectButtons onSelect={(matches)=>this.selectMatchingStudies(matches)}
```

Comment:
doesnt matter very much but you could autobind this.selectedMatchingStudies and just pass that in